src/Components/WebAssembly/JSInterop/src/WebAssemblyJSRuntime.cs

```diff
@@ -102,13 +101,29 @@ protected override void SendByteArray(int id, byte[] data)
                         ? throw new JSException(exception)
                         : (TResult)(object)new WebAssemblyJSObjectReference(this, id);
                 case JSCallResultType.JSStreamReference:
-                    // WebAssembly streaming interop is not presently supported (TODO). Thus we shouldn't see
-                    // a JSCallResultType.JSStreamReference.
+                    var serializedStreamReference = InternalCalls.InvokeJS<T0, T1, T2, string>(out exception, ref callInfo, arg0, arg1, arg2);
+                    return exception != null
+                        ? throw new JSException(exception)
+                        : (TResult)(object)DeserializeJSStreamReference(serializedStreamReference);
                 default:
                     throw new InvalidOperationException($"Invalid result type '{resultType}'.");
             }
         }
 
+        private IJSStreamReference DeserializeJSStreamReference(string serializedStreamReference)
+        {
+            var receivedBytes = Encoding.UTF8.GetBytes(serializedStreamReference);
+            var reader = new Utf8JsonReader(receivedBytes);
+            var jsStreamReference = JsonSerializer.Deserialize<IJSStreamReference>(ref reader, JsonSerializerOptions);
```

Comment:
Why can't you call `JsonSerializer.Deserialize<IJSStreamReference>(serializedStreamReference, JsonSerializerOptions);`?